src/main/java/executor/service/handler/TerminatorListener.java

```diff
@@ -0,0 +1,5 @@
+package executor.service.handler;
+
+public interface TerminatorListener {
```

Comment:
Explain why you are coding this code and trying to make changes to the already killed code in the main branch. If you need this for tests, to change the logic of the while loop, then you can use Reflection (**java.lang.reflect**). An example of this approach can be seen in ParallelFlowExecutorServiceImplTest [#32](https://github.com/GFLCourses5/worker/pull/32). Thanks

Reply:
Not sure what do you mean under "trying to make changes to the already killed code in the main branch". Please provide me more clear and detailed explanation.

Reply:
Sorry, into already approved code, beta-main branch.

Reply:
The proposed code does not change *ScenarioListener* logic. The only purpose is to perform testing. And I don't want to put additional flags and use reflection.